### ISSUE TITLE
epochs datetime missing

### ISSUE THREAD

**Issue by AndrewLouw** [brandon-rhodes/python-sgp4#62]
The sgp4 model used to have a property .epoch that returned a python datetime object. This seems to have been removed. It's also been removed from releases of 2.3 so it's not available if I install a past version. There doesn't seem to be any function for getting a datetime from the jsatepoch and jsatepochF. Sorry if the issue is at my end

**Comment by brandon-rhodes** [comment]
> It's also been removed from releases of 2.3 so it's not available if I install a past version.

Past versions should work like they always have, and should offer a `Satellite` object with the traditional `epoch` attribute. And if not they should be fixed — could you outline how you are installing the older version, and then show the output you are getting when you try using the old `epoch` attribute?

You are correct that the new C++ accelerated `Satrec` does not provide any Python objects, only simple floating point numbers, for the epoch field. I will look into updating the documentation for folks who want the epoch field turned into a datetime. In the meantime, I think this should work for you:

```
from sgp4.ext import invjday
year, mon, day, hr, minute, sec = invjday(sat.jdsatepoch + sat.jdsatepochF)
```

You can then use those 6 values to build a datetime.

**Comment by AndrewLouw** [comment]
I've been installing with:
pip install sgp4==2.3

The code I've been running is essentially:

````python
        from sgp4.api import Satrec
        model = Satrec.twoline2rv(line1, line2)
        satnum = model.satnum
        epoch= model.epoch
````
I have a  virtual environment from earlier this year that is running 2.3 (according to pip freeze) and this block works there, but not on my cloud virtual machine that I created this week and installed with: 
pip install sgp4==2.3 

(the working version is windows, the VM is Linux - but that shouldn't make a difference)

My only guess is that there was an update to 2.3 that didn't increment version number. And this happened after I created my virtual environment. 

I've been using:
        seconds = self.model.jdsatepoch + self.model.jdsatepoch
        epoch = datetime.datetime.utcfromtimestamp(seconds)
Any reason this shouldn't work too?
(I've been getting values, but I don't know if they're right)

Hope this helps, thanks

**Comment by brandon-rhodes** [comment]
> My only guess is that there was an update to 2.3 that didn't increment version number.

The Python Package Index, if I recall, disallowed updates to packages many years ago — so I am not sure it would have let me upload a new 2.3 version.

There is another possibility, though: if `sgp4` can't compile its C++ backend, then it falls back to (slowly) implementing the calculations with Python. When it does so, it uses the old code that sets the `epoch` datetime. Check for whether `sgp4.api.accelerated` is `True` in one of your environments and `False` in the other.

And in addition to documenting a new and better way to get an epoch datetime, I should probably make the fallback code delete that attribute so that it never misleads any programmers who discover that it's there when the C++ fails to compile and expect it to be there later when maybe it does!

**Comment by brandon-rhodes** [comment]
There! A new function `conveniences.sat_epoch_datetime()` will be included in the next version of Skyfield to support epoch datetimes against the new C++ powered `Satrec` class. Feel free to test it out ahead of the next release:
```
pip install https://github.com/brandon-rhodes/python-sgp4/archive/master.zip
```
Simply comment here if you have any trouble with it.

**Comment by brandon-rhodes** [comment]
I have just released `sgp4` 2.9 that includes the new epoch routines. Thanks for pointing out that there needed to be a new way to reduce them to Python `datetime` objects!

**Comment by AndrewLouw** [comment]
Hi, thanks for the quick implementation of this, It's a big help :)
